src/client/components/multi-form-editor/landing-page.js

```diff
@@ -1,36 +1,224 @@
 const h = require('react-hyperscript');
 const { Link } = require('react-router-dom');
 const React = require('react');
+const ReactDom = require('react-dom');
+const { makeClassList } = require('../../../util');
+const Promise = require('bluebird');
+const Linkout = require('../document-linkout');
+const anime = require('animejs');
 
 class LandingPage extends React.Component {
+  constructor( props ){
+    super( props );
+
+    this.state = {
+      textAreaEnabled: false,
+      submitting: false,
+      textAreaAnimating: false,
+      textAreaHeight: '20em'
```

Comment:
Please don't mix styles with JS. Use CSS instead.

Reply:
I define CSS values in CSS files and use adding/removing classes for changing them dynamically. However, this is a different case. As you most probably realized the text area is opened and closed by animating through its height. Therefore I need to store its height before it is closed to use when it is opened back (Note that the height of text area can be changed by users). Here is the code line where how the height is supposed the change during animation is defined (https://github.com/metincansiper/factoid/blob/landing-page-ui/src/client/components/multi-form-editor/landing-page.js#L107). While opening the text area it should animate between 0 and the height of text area that is saved before closing it. And the initial value for the open height (I mean the height that it has when it is open) of it is defined at the code segment that you highlighted. If I defined the initial height of text area through CSS, it would be overridden during the animation. Therefore, this is the way that sounds best to me. If you suggest a better way I will be glad to hear and discuss.

Reply:
MMhh... I understand but can't you toggle-add a css class with the height when opened and use animation on height with ease on the parent css class? A small most likely not working, offline written example to illustrate the idea:

```javascript
<element className={this.state.textAreaActive===false ? "textArea" : "textArea open" }
```

```css
.textArea {
   ....;
   max-height: 0;
   transition: height 1s;
  overflow: hidden;
}
.textArea > .open {
   max-height: 6em;
}
```

Or: https://codepen.io/LFeh/pen/ICkwe

Reply:
Yes @sacdallago your approach is the convention that this code base uses for state based css

Reply:
@sacdallago Thank you for the suggestion. However, I have 2 main concerns about it. One is more important while the other one would not be that much important.

The important concern of me is determining a value for ```max-height``` for the text area when it is open (I mean the exact value to be written instead of ```max-height: 6em;``` in your example). As I mentioned users can increase the height of text area. Therefore, we do not know what would be the maximum height that it should have when it is open. In this case one would think of setting an arbitrary large value for the ```max-height``` however, in that case the animation would be destroyed. I mean in case of ```max-height``` is much more than ```height``` users will observe that the accordion will open immediately with nearly no transition effect while opening the text area. While closing it they will observe that there will be no effect for a while (until the ```max-height``` become equal to ```height```) then again it will be closed immediately with nearly no transition effect. Also, even the ```max-height``` that we preset would not be enough. 

The minor concern is that I need to do something before the animation in case text area is being opened and do it after the animation is completed while it is being closed (You can see https://github.com/metincansiper/factoid/blob/landing-page-ui/src/client/components/multi-form-editor/landing-page.js#L116). For this reason ```promise``` support of ```animejs``` library was a good facility. I can try to use ```transitionend``` event instead but I suspect that it would decrease the readability and would require using some not preferred practices. 

As a final comment, your concern about the initial implementation of me was that I was keeping a property in state object that is assigned to```20em``` initially and you considered it as mixing JS with styles. However, from a different perspective it just keeps a state (saying what should the height of text area be) to be used as a parameter while having a call of ```animejs``` library.

What do you think about all of these?

@d2fong I do not agree that approach of @sacdallago was strictly on use in this code base for similar animations. You can see https://github.com/PathwayCommons/factoid/blob/alpha/src/client/components/document-filler.js#L45. Actually, I needed to move this code segment to landing page since document filler's content is moved to there. That code segment would be totally removed according to the discussion on https://github.com/PathwayCommons/factoid/issues/176. However, it would needed to be updated accordingly in case it will not be removed.